theano/d3viz/formatting.py

```diff
@@ -0,0 +1,343 @@
+"""Functions for formatting Theano compute graphs.
+
+Author: Christof Angermueller <cangermueller@gmail.com>
+"""
+
+import numpy as np
+import logging
+import os.path as pt
+from functools import reduce
+
+try:
+    import pydot as pd
+    if pd.find_graphviz():
+        pydot_imported = True
+    else:
+        pydot_imported = False
+except ImportError:
+    pydot_imported = False
+
+import theano
+from theano import gof
+from theano.compile.profilemode import ProfileMode
+from theano.compile import Function
+from theano.compile import builders
+
+_logger = logging.getLogger("theano.printing")
+
+
+class PyDotFormatter(object):
+
+    def __init__(self, compact=True):
+        """Converts compute to to `pydot` object.
+
+        :param compact: if True, will remove intermediate variables without
+                        name.
+        """
+
+        if not pydot_imported:
+            raise RuntimeError("Failed to import pydot. You must install pydot"
+                               " for `pydotprint` to work.")
```

Comment:
This will lead to a strange error fo people calling d3write() and getting an error about pydotprint().